File: 06-Code/RolePaymentSystem_V2.0_Python/model/Employee.py
from datetime import date
class Employee:
    def __init__(self, name, last_name, id_number, hire_date, basic_salary, overtime_hours, absent_days, bonuses, iess_loans, company_loans, fines, bring_own_food):
        self.name = name
        self.last_name = last_name
        self.id_number = id_number
        self.hire_date = hire_date
        self.basic_salary = basic_salary
        self.overtime_hours = overtime_hours
        self.absent_days = absent_days
        self.bonuses = bonuses
        self.iess_loans = iess_loans
        self.company_loans = company_loans
        self.fines = fines
        self.bring_own_food = bring_own_food
        
    def to_dict(self):
        return {
            'name': self.name,
            'last_name': self.last_name,
            'id_number': self.id_number,
            'hire_date': self.hire_date.isoformat() if isinstance(self.hire_date, date) else str(self.hire_date),
            'basic_salary': self.basic_salary,
            'overtime_hours': self.overtime_hours,
            'absent_days': self.absent_days,
            'bonuses': self.bonuses,
            'iess_loans': self.iess_loans,
            'company_loans': self.company_loans,
            'fines': self.fines,
            'bring_own_food': self.bring_own_food
        }
    def get_name(self):
        return self.name

    def set_name(self, name):
        self.name = name

    def get_last_name(self):
        return self.last_name

    def set_last_name(self, last_name):
        self.last_name = last_name

    def get_id_number(self):
        return self.id_number

    def set_id_number(self, id_number):
        self.id_number = id_number

    def get_hire_date(self):
        return self.hire_date

    def set_hire_date(self, hire_date):
        self.hire_date = hire_date

    def get_basic_salary(self):
        return self.basic_salary

    def set_basic_salary(self, basic_salary):
        self.basic_salary = basic_salary

    def get_overtime_hours(self):
        return self.overtime_hours

    def set_overtime_hours(self, overtime_hours):
        self.overtime_hours = overtime_hours

    def get_absent_days(self):
        return self.absent_days

    def set_absent_days(self, absent_days):
        self.absent_days = absent_days

    def get_bonuses(self):
        return self.bonuses

    def set_bonuses(self, bonuses):
        self.bonuses = bonuses

    def get_iess_loans(self):
        return self.iess_loans

    def set_iess_loans(self, iess_loans):
        self.iess_loans = iess_loans

    def get_company_loans(self):
        return self.company_loans

    def set_company_loans(self, company_loans):
        self.company_loans = company_loans

    def get_fines(self):
        return self.fines

    def set_fines(self, fines):
        self.fines = fines

    def is_bring_own_food(self):
        return self.bring_own_food

    def set_bring_own_food(self, bring_own_food):
        self.bring_own_food = bring_own_food

File: 06-Code/Graphic interface/GraphicInterface/src/main/java/ec/espe/edu/rolepaymentsystem/model/Employee.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.espe.edu.rolepaymentsystem.model;

import java.util.Date;
import java.util.Objects;

/**
 *
 * @author Code Master
 */
public class Employee {
    private String name;
    private String lastName;
    private String idNumber;
    private Date hireDate;
    private double basicSalary;
    private double overtimeHours;
    private double absentDays;
    private double bonuses;
    private double iessLoans;
    private double companyLoans;
    private double fines;
    private boolean bringOwnFood;

    public Employee(String name, String lastName, String idNumber, Date hireDate, double basicSalary, double overtimeHours, double absentDays, double bonuses, double iessLoans, double companyLoans, double fines, boolean bringOwnFood) {
        this.name = name;
        this.lastName = lastName;
        this.idNumber = idNumber;
        this.hireDate = hireDate;
        this.basicSalary = basicSalary;
        this.overtimeHours = overtimeHours;
        this.absentDays = absentDays;
        this.bonuses = bonuses;
        this.iessLoans = iessLoans;
        this.companyLoans = companyLoans;
        this.fines = fines;
        this.bringOwnFood = bringOwnFood;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getIdNumber() {
        return idNumber;
    }

    public void setIdNumber(String idNumber) {
        this.idNumber = idNumber;
    }

    public Date getHireDate() {
        return hireDate;
    }

    public void setHireDate(Date hireDate) {
        this.hireDate = hireDate;
    }

    public double getBasicSalary() {
        return basicSalary;
    }

    public void setBasicSalary(double basicSalary) {
        this.basicSalary = basicSalary;
    }

    public double getOvertimeHours() {
        return overtimeHours;
    }

    public void setOvertimeHours(double overtimeHours) {
        this.overtimeHours = overtimeHours;
    }

    public double getAbsentDays() {
        return absentDays;
    }

    public void setAbsentDays(double absentDays) {
        this.absentDays = absentDays;
    }

    public double getBonuses() {
        return bonuses;
    }

    public void setBonuses(double bonuses) {
        this.bonuses = bonuses;
    }

    public double getIessLoans() {
        return iessLoans;
    }

    public void setIessLoans(double iessLoans) {
        this.iessLoans = iessLoans;
    }

    public double getCompanyLoans() {
        return companyLoans;
    }

    public void setCompanyLoans(double companyLoans) {
        this.companyLoans = companyLoans;
    }

    public double getFines() {
        return fines;
    }

    public void setFines(double fines) {
        this.fines = fines;
    }

    public boolean getBringOwnFood() {
        return bringOwnFood;
    }

    public void setBringOwnFood(boolean bringOwnFood) {
        this.bringOwnFood = bringOwnFood;
    }
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Employee employee = (Employee) o;
        return Objects.equals(idNumber, employee.idNumber);
    }

    @Override
    public int hashCode() {
        return Objects.hash(idNumber);
    }
}
